python_modules/dagster/dagster/_utils/caching_instance_queryer.py

```diff
@@ -581,9 +581,12 @@ def get_asset_partitions_updated_after_cursor(
             for asset_partition, latest_storage_id in self._get_latest_materialization_or_observation_storage_ids_by_asset_partition(
                 asset_key=asset_key
             ).items()
-            if (asset_partitions is None or asset_partition in asset_partitions)
-            and (latest_storage_id or 0) > (after_cursor or 0)
+            if (latest_storage_id or 0) > (after_cursor or 0)
         }
+
+        if asset_partitions:
```

Comment:
we should never be passing in an empty list here but I always prefer to be specific with these comparisons (if asset_partitions is not None)